Sources/Core/AWSClientRuntime/Endpoints/AWSEndpoint.swift

```diff
@@ -58,6 +58,11 @@ public struct AWSEndpoint: Equatable {
 }
 
 extension Endpoint {
+
+    /// Returns the auth scheme for the given name
+    /// This is an internal API and subject to change without notice
```

Comment:
Not for this PR: perhaps we should prefix internal API's to make that more clear and consistent. Such as with a `_` like `_authScheme`

Reply:
This is interesting suggestion but my worry is, anything with _ prefix is used for private access only. Having something with _ and public would make things harder for reviews.

Rather I would like to have something like @internal attribute which is more explicit. Sadly, Swift doesn't have custom attribute support yet.

https://forums.swift.org/t/pitch-introduce-custom-attributes/21335/21

Reply:
Unfortunately that proposal doesn't have much traction these days and there doesn't appear to be a timeline of when, or even if, it will get completed.

We don't necessarily need to use a preceding `_`, but i would like us to come up with a convention that will make it easy for us to identify cases like this when reading code. Otherwise imo we should remove the "for internal use only comment" as there is nothing preventing non-internal use.

Reply:
no matter what, nothing stops customers to use this method, it is public.

Intent is to communicate. Either via a comment, an attribute or something else. So in future if just remove this, we can say that this was expected.

Most SDKs struggle with this, there is no good way to share code without making it public. Some namespace it with `internal`, some add attribute or some add comment.

Reply:
More fundamentally, is there a reason we need this disclaimer on this function?  The function is documented and does a pretty straightforward thing.  Why treat it different from any other public function?

Reply:
that's because of nature of data it deals with. EndpointRuleSet are expected to evolve and setting an expectation on a dynamic data is not good.

We have had trouble in past, for example in some SDKs we support giving list of supported region. This seems fair at high level but gets complicated to maintain. For example, if the source of data goes away and replaced by something that is not compatible.

Reply:
I will approve this because it follows the existing pattern in the project, but we should revisit it.  The fact that we feel the need to declare something "public but not public", modularization or not, suggests that something is wrong with the patterns we are employing.

I think doing a better job of encapsulating these fields & data in types meant to hold and work with them will prevent us from having to make these types of declarations and make generated code simpler since we won't have to pick through individual data fields when configuring reusable types like middleware.